src/Eagle.ts

```diff
@@ -2104,14 +2110,23 @@ export class Eagle {
     addNodeToLogicalGraph = (node : Node) : void => {
         //console.log("addNodeToLogicalGraph()", node.getName(), node.getCategory(), node.getInputPorts().length, node.getOutputPorts().length, node.getFields().length);
 
+        
         // get new position for node
-        let pos = this.getNewNodePosition();
+        if (Eagle.nodeDropLocation.x == 0 && Eagle.nodeDropLocation.y == 0){
+            Eagle.pos = this.getNewNodePosition();
+        }else if (Eagle.nodeDropLocation){
+            Eagle.pos = Eagle.nodeDropLocation;
+        }else{
+            Eagle.pos = {x:0, y:0};
+            alert("Unexpected error occurred")
```

Comment:
I usually don't use alert() for situations like this, except during development.

I'd recommend using Utils.showNotification() if you want to notify the user of error like this.